airflow/settings.py

```diff
@@ -71,6 +74,43 @@ def timing(cls, stat, dt):
 
 Stats = DummyStatsLogger
 
+
+ALLOWED_STAT_CHAR = set(string.ascii_letters + string.digits + '_.')
```

Comment:
Do we want to also include `-` as [we allow it in dag_id](https://github.com/apache/airflow/blob/717ddaf737ec1c57244377f73dd0e80e1628576a/airflow/utils/helpers.py#L48)? Maybe it is out of the scope but would be nice if we can have one source of truth REGEX for allowed char.

Reply:
Yup we should reuse what we already have for validation.